airlock/business_logic.py

```diff
@@ -276,6 +279,29 @@ def get_url(self, relpath: UrlPath = ROOT_PATH) -> str:
             kwargs["path"] = str(relpath)
         return reverse("workspace_view", kwargs=kwargs)
 
+    def get_workspace_state(self, relpath: UrlPath) -> WorkspaceFileState | None:
+        try:
+            self.abspath(relpath)
+        except BusinessLogicLayer.FileNotFound:
```

Comment:
Why would a file not be found?  This could do with a comment.

Reply:
"Shit happens" :rofl: 

Seriously, was just defense in depth so we don't blow up if we get passed a bad path.

Reply:
Blowing up here would mean the entire workspace cannot be viewed, as this is called when rendering the tree,

But it is a very unlikely case, so maybe that's ok.